### ISSUE TITLE
Toast has no accessible name due to missing aria-labelledby

### ISSUE THREAD

**Issue by stefee** [chakra-ui/chakra-ui#4699]
### Description

The toast component renders a `role="alert"` without `aria-labelledby` attribute (or similar) which means the accessible name of the alert is read as empty string.

### Link to Reproduction

https://codesandbox.io/s/serene-mcnulty-d7qg8?file=/src/index.js

### Steps to reproduce

```js
const toast = useToast();

return (
  <Button
    onClick={() => {
      toast({
        title: 'My Toast Title',
      });
    }}
  >
    Show Toast
  </Button>
);
```

Renders this to the document:

```html
<div class="chakra-alert css-1kao88d" id="2" role="alert">
  <span class="chakra-alert__icon css-1c6vscd">
    <svg class="chakra-icon css-fi0ww0" focusable="false" viewBox="0 0 24 24">
      <path
        d="M12,0A12,12,0,1,0,24,12,12.013,12.013,0,0,0,12,0Zm.25,5a1.5,1.5,0,1,1-1.5,1.5A1.5,1.5,0,0,1,12.25,5ZM14.5,18.5h-4a1,1,0,0,1,0-2h.75a.25.25,0,0,0,.25-.25v-4.5a.25.25,0,0,0-.25-.25H10.5a1,1,0,0,1,0-2h1a2,2,0,0,1,2,2v4.75a.25.25,0,0,0,.25.25h.75a1,1,0,1,1,0,2Z"
        fill="currentColor" />
    </svg>
  </span>
  <div class="css-njbp03">
    <div class="chakra-alert__title css-0">
      My Toast Title
    </div>
  </div>
</div>
```

The parent `<div>` with `role="alert"` does not have `aria-labelledby`. The child div containing text `My Toast Title` does not have an `id` which can be referenced via `aria-labelledby` on the parent.

The [accessible name can be found using browser inspector](https://knowledge.evinced.com/concepts/accessible-names).

<img width="1552" alt="Google Chrome inspector screenshot, highlighting computer name is not specified" src="https://user-images.githubusercontent.com/18026180/132857190-44a3f571-2109-4f80-bc70-bc6e1d3214ee.png">

### Chakra UI Version

1.6.7

### Browser

Google Chrome

### Operating System

- [X] macOS
- [ ] Windows
- [ ] Linux

### Additional Information

W3C spec for `alert` role: https://www.w3.org/TR/wai-aria/#alert
W3C spec for 'aria-labelledby': https://www.w3.org/TR/wai-aria/#aria-labelledby
W3C spec for Accessible Name and Description: https://www.w3.org/TR/accname-1.1/#mapping_additional_nd

Note that the `alert` role has the following property:

> Name From: author

This means that the accessible name will not be inferred from text content of the alert, and so an additional attribute must be provided for the alert to have an accessible name.

**Comment by stefee** [comment]
We discovered this bug during unit testing because we are using react-testing-library and following this pattern: https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#not-using-byrole-most-of-the-time

For example, to find a toast in the document after a user event is triggered:

> await screen.findByRole('alert', { name: /My Toast Title/i });

The above example does not currently work, because the toast alert div does not have an accessible name in the tree, and so react-testing-library is not able to find the toast alert by its accessible name.

Instead, we must do this:

> await screen.findByText(/My Toast Title/i);

**Comment by stefee** [comment]
After doing some more reading, I am now unsure if our assumption that alerts should be discoverable by an accessible name is correct.

Here is a quote from [WAI-ARIA Authoring Practices - Accessible Name Guidance By Role](https://w3c.github.io/aria-practices/#naming_role_guidance):

> Some screen readers announce the name of an alert before announcing the content of the alert. Thus, aria-label provides a method for prefacing the visible content of an alert with text that is not displayed as part of the alert. Using aria-label is functionally equivalent to providing off-screen text in the contents of the alert, except off-screen text would be announced by screen readers that do not support aria-label on alert elements.

I interpret this sentence to mean that an accessible name is only encouraged on alerts if the title is _not_ already present in the content of the alert, because some screen readers will announce the name and then also announce the content, so using both will cause a repeated announcement.

However, this guidance has been disputed here: https://github.com/w3c/aria-practices/issues/1218

My general feeling now is that the way Chakra UI is currently handling alert title is already presenting the best outcome for screen readers, but it only presents a challenge from an automated testing point of view. I think what I would recommend is to keep the ARIA tree the same as it is now, but allow a way of passing a `data-testid` to the toast component via `useToast` or `createStandaloneToast` for testing purposes.

**Comment by stefee** [comment]
I have done some more experimentation using screen readers and I have found that `aria-labelledby` behaves how we want it to behave (it only reads the title once) whereas `aria-label` will cause the title to be read twice (once for the label, again for the content).

https://codesandbox.io/s/serene-mcnulty-d7qg8?file=/src/index.js

I have updated this reproduction linked in the original description to include an example of each of the following:

- Button which creates a default toast
- Button which creates a toast without aria attributes (similar to default)
- Button which creates a toast which has `aria-labelledby`
- Button which creates a toast which has `aria-label`

With all of the above in mind, I think `aria-labelledby` is the best approach because it solves the automated testing problem and also provides best outcome for screen readers.
